PurchasesCoreSwift/Misc/SystemInfo.swift

```diff
@@ -0,0 +1,195 @@
+    }
+
+    @objc public static func systemVersion() -> String {
+        return ProcessInfo().operatingSystemVersionString
+    }
+
+    @objc public static func appVersion() -> String {
+        return Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? ""
+    }
+
+    @objc public static func buildVersion() -> String {
+        return Bundle.main.infoDictionary?["CFBundleVersion"] as? String ?? ""
+    }
+
+    @objc public static func platformHeader() -> String {
+        return Self.forceUniversalAppStore ? "iOS" : PLATFORM_HEADER
+    }
+
+    @objc public static func identifierForVendor() -> String? {
+        #if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
+            return UIDevice.current.identifierForVendor?.uuidString
+        #elseif os(watchOS)
+            return WKInterfaceDevice.current().identifierForVendor?.uuidString
+        #else
+            return nil
+        #endif
+    }
+
+    private static var privateProxyURL: URL?
+    @objc public static func proxyURL() -> URL? {
+        return privateProxyURL
+    }
+
+    @objc public static func setProxyURL(_ proxyURL: URL?) {
+        privateProxyURL = proxyURL
+        guard let privateProxyURLString = privateProxyURL?.absoluteString else {
+            return
+        }
+        Logger.info(ConfigureStrings.configuring_purchases_proxy_url_set.replacingOccurrences(of: "%@",
+                                                                                              with: privateProxyURLString))
+    }
+
+    static func defaultServerHostURL() -> URL {
+        return URL(string: defaultServerHostName)!
+    }
+
+    @objc public static func serverHostURL() -> URL {
+        return proxyURL() ?? Self.defaultServerHostURL()
+    }
+
+    private var isApplicationBackgrounded: Bool {
+    #if os(iOS) || (targetEnvironment(simulator) && os(iOS))
+        return self.isApplicationBackgroundedIOS
+    #elseif os(tvOS)
+    return UIApplication.shared.applicationState == UIApplication.State.background
+    #elseif os(macOS)
+        return false
+    #elseif os(watchOS)
+        return  WKExtension.shared().applicationState == WKApplicationState.background
+    #endif
+    }
+
+    #if os(iOS)
+    // iOS App extensions can't access UIApplication.sharedApplication, and will fail to compile if any calls to
+    // it are made. There are no pre-processor macros available to check if the code is running in an app extension,
+    // so we check if we're running in an app extension at runtime, and if not, we use KVC to call sharedApplication.
+
+    @objc private var isApplicationBackgroundedIOS: Bool {
+        if self.isAppExtension {
+            return true
+        }
+
+        return UIApplication.shared.applicationState == UIApplication.State.background
+    }
+
+    @objc private var isAppExtension: Bool {
+        return Bundle.main.bundlePath.hasSuffix(".appex")
+    }
+
+    #endif
+
+    @available(*, deprecated, message: """
```

Comment:
instead of crashing, we new pass an error back. This simplified testing, since M1 macs can test the particular crashes due to exception handling particularities.

Reply:
this class isn't public, so I think instead of deprecating we should just remove the old initializer and update the code to use the new one

Reply:
Unfortunately, the class is public, the API tester can see it by `@import Purchases;` only.

Reply:
Discussed with @aboedo and did some more digging, it isn't public 🎉, updating